### ISSUE TITLE
Warnings in trace with Intel 18 compilers

### ISSUE THREAD

**Issue by lifflander** [DARMA-tasking/vt#647]
**Describe the bug**
```
/lscratch1/jliffla/vt-clean/vt-auto-build/vt-release-new-1b/vt/vt/src/vt/trace/trace.cc(801): warning #181: argument of type "vt::trace::TraceEntrySeqType={uint32_t={unsigned int}}" is incompatible with format "%lu", expecting argument of type "unsigned long"
          event_seq_id,
          ^

/lscratch1/jliffla/vt-clean/vt-auto-build/vt-release-new-1b/vt/vt/src/vt/trace/trace.cc(805): warning #181: argument of type "vt::trace::TraceMsgLenType={size_t={unsigned long}}" is incompatible with format "%d", expecting argument of type "int"
          log->msg_len,
          ^

/lscratch1/jliffla/vt-clean/vt-auto-build/vt-release-new-1b/vt/vt/src/vt/trace/trace.cc(818): warning #181: argument of type "vt::trace::TraceEntrySeqType={uint32_t={unsigned int}}" is incompatible with format "%lu", expecting argument of type "unsigned long"
          event_seq_id,
          ^

/lscratch1/jliffla/vt-clean/vt-auto-build/vt-release-new-1b/vt/vt/src/vt/trace/trace.cc(822): warning #181: argument of type "vt::trace::TraceMsgLenType={size_t={unsigned long}}" is incompatible with format "%d", expecting argument of type "int"
          log->msg_len,
          ^

/lscratch1/jliffla/vt-clean/vt-auto-build/vt-release-new-1b/vt/vt/src/vt/trace/trace.cc(853): warning #181: argument of type "vt::trace::TraceEntrySeqType={uint32_t={unsigned int}}" is incompatible with format "%lu", expecting argument of type "unsigned long"
          event_seq_id,
          ^

/lscratch1/jliffla/vt-clean/vt-auto-build/vt-release-new-1b/vt/vt/src/vt/trace/trace.cc(857): warning #181: argument of type "vt::trace::TraceMsgLenType={size_t={unsigned long}}" is incompatible with format "%d", expecting argument of type "int"
          log->msg_len,
          ^

/lscratch1/jliffla/vt-clean/vt-auto-build/vt-release-new-1b/vt/vt/src/vt/trace/trace.cc(868): warning #181: argument of type "vt::trace::TraceEntrySeqType={uint32_t={unsigned int}}" is incompatible with format "%lu", expecting argument of type "unsigned long"
          event_seq_id,
          ^

/lscratch1/jliffla/vt-clean/vt-auto-build/vt-release-new-1b/vt/vt/src/vt/trace/trace.cc(872): warning #181: argument of type "vt::trace::TraceMsgLenType={size_t={unsigned long}}" is incompatible with format "%d", expecting argument of type "int"
          log->msg_len
          ^

```

**Comment by pnstickne** [comment]
Sweet, this was the type widening issue previously mentioned.. I'll take.

**Comment by pnstickne** [comment]
Closed with #747 merge.
